src/coreclr/src/jit/lsra.cpp

```diff
@@ -5454,10 +5486,109 @@ void LinearScan::allocateRegisters()
+                    Interval*  assignedInterval = physRegRecord->assignedInterval;
+                    if (assignedInterval != nullptr)
+                    {
+                        bool         isAssignedReg     = (assignedInterval->physReg == reg);
+                        RefPosition* recentRefPosition = assignedInterval->recentRefPosition;
+                        // If we have a copyReg or a moveReg, we might have assigned this register to an Interval,
+                        // but that isn't considered its assignedReg.
+                        if (recentRefPosition != nullptr)
+                        {
+                            if (recentRefPosition->refType == RefTypeExpUse)
+                            {
+                                // We don't update anything on these, as they're just placeholders to extend the
+                                // lifetime.
+                                continue;
+                            }
+                            // For copyReg or moveReg, we don't have anything further to assert.
+                            if (recentRefPosition->copyReg || recentRefPosition->moveReg)
+                            {
+                                continue;
+                            }
+                            assert(assignedInterval->isConstant == isRegConstant(reg, assignedInterval->registerType));
+                            if (assignedInterval->isActive)
+                            {
+                                // If this is not the register most recently allocated, it must be from a copyReg,
+                                // or it was placed there by the inVarToRegMap. In either case it must be a lclVar.
+
+                                if (!isAssignedToInterval(assignedInterval, physRegRecord))
+                                {
+                                    assert(assignedInterval->isLocalVar);
+                                    // We'd like to assert that this was either set by the inVarToRegMap, or by
+                                    // a copyReg, but we can't traverse backward to check for a copyReg, because
+                                    // we only have recentRefPosition, and there may be a previous RefPosition
+                                    // at the same Location with a copyReg.
+                                }
+                                if (isAssignedReg)
+                                {
+                                    assert(nextIntervalRef[reg] == assignedInterval->getNextRefLocation());
+                                    if (assignedInterval->isActive)
+                                    {
+                                        assert(!isRegAvailable(reg, assignedInterval->registerType));
+                                        assert((recentRefPosition == nullptr) ||
+                                               (spillCost[reg] == getSpillWeight(physRegRecord)));
+                                    }
```

Comment:
This block will always be true because you check for `if (assignedInterval->isActive)` above on line 5529. Perhaps you want an `else` counterpart somewhere below which will check this portion for `!assignedInterval->isActive`?

```c++
if (!isAssignedToInterval(assignedInterval, physRegRecord))
{
    assert(assignedInterval->isLocalVar);
    // We'd like to assert that this was either set by the inVarToRegMap, or by
    // a copyReg, but we can't traverse backward to check for a copyReg, because
    // we only have recentRefPosition, and there may be a previous RefPosition
    // at the same Location with a copyReg.
}
if (isAssignedReg)
{
    assert(nextIntervalRef[reg] == assignedInterval->getNextRefLocation());
    assert(isRegAvailable(reg, assignedInterval->registerType));
    assert(spillCost[reg] == 0);
}
```

Reply:
Yes, I think this belongs below.